tailcfg/tailcfg.go

```diff
@@ -875,6 +875,8 @@ type Hostinfo struct {
 	// explicitly declared by a node.
 	Location *Location `json:",omitempty"`
 
+	TPMAvailable opt.Bool // set to true if a local TPM device is available
```

Comment:
you don't want any other metadata? version? type? capabilities? bool is really sufficient?

Reply:
2.0 is the version we will support and that's what this bool checks for (it will be false for 1.2).
there aren't any special capabilities that we need (just sealing for now, which is always supported), and I don't know of a way to query capabilities from a TPM device other than trying all the commands. there isn't really that much variability in TPM capabilities AFAIK.
so yeah, I think a bool is sufficient here.

Reply:
actually, I take that back, there's https://pkg.go.dev/github.com/google/go-tpm/tpm2#GetCapability which can return some of https://pkg.go.dev/github.com/google/go-tpm/tpm2#TPMPT.
getting vendor/model info might help us with debugging later. I'll update the PR.

Reply:
And then rename TPMAvailable to just TPM?

Reply:
Yep, now it's a struct with some useful-looking device info. PTAL!